### PR TITLE
Function5and6

### PR DESCRIPTION
**Pull request by ZachBowman**

### CODE REVIEW

File: parse_places_array.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  var places_data = new Array();
+
+  var total_places = 0;  // total places we've parsed into the object array
+  
+  // we need a two-dimensional loop to traverse the array
+  for (var request_index = 0; request_index < json_array.length; request_index += 1)
+    {
+    for (var place_index = 0; place_index < json_array[request_index].length; place_index += 1)
+      {
+      places_data[total_places] = JSON.parse (json_array[json_index]);
```
Comment (AndyCErnst) [inline parse_places_array.js:19]: could be changed to a push, like: places_data.push( JSON.parse (json_array[json_index]));
Fewer moving parts that way.

File: parse_places_array.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// json_array will be a 2-dimensional array, with each cell containing a json string.
+// The outside dimension will have one row for each buffer request (bounding box) returned
+// by google places.
+// Each row will contain one cell for each place returned by that request.
+// Each cell will contain one json string, representing one place.
+
+function parse_places_array (json_array)
+  {
+  // we must create an array of objects to hold the result data
+  var places_data = new Array();
```
Comment (AndyCErnst) [inline parse_places_array.js:10]: places_data will stop existing once this function exits. Maybe we should return that object instead and do a .length on it if we need the length?

File: plot_places [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  var marker = new Array();  // the array to hold the map marker data
+  var total_markers = 0;     // total markers in our array so far
+
+  // run through each place, saving the relevent data to a new marker as we go
+  for (var place_index = 0; place_index < places_data.length; place_index += 1)
+    {
+    var place = places_data[place_index];  // for less typing
+    var m = total_markers;                 // also for less typing
+
+    marker[m] = new google.maps.Marker();
```
Comment (AndyCErnst) [inline plot_places:15]: I just looked this up, Markers must have their position set in their constructor. Here is the site example:
<code>
var marker = new google.maps.Marker({
    position: myLatlng, 
    map: map, 
    title:"Hello World!"
});
</code>
Position is required, map is optional but needed to display on map, title is optional

**Comment by kevinmstephens** [comment]
We need to rethink these function in light of how the places will be acquired (async.). Each bounding box request can be passed a callback function that will be passed a results object and a pagination object. The results can be parsed and the pagination can be used to figure out if there are more than 20 results returned.

File: plot_places (hunk 3 above)
Comment (ZachBowman) [inline plot_places:15]: I couldn't find a spot where it says it has to be in the constructor, but the site example does show it that way, and I should probably change it just to be safe.

File: parse_places_array.js (hunk 2 above)
Comment (ZachBowman) [inline parse_places_array.js:10]: Oh, this is my mistake.  I returned total_places - it should be "return places_data;".
Thanks Andy.

File: plot_places (hunk 3 above)
Comment (AndyCErnst) [inline plot_places:15]: Yeah, this https://developers.google.com/maps/documentation/javascript/markers#add gets very close to saying it must be in the constructor. I interpreted it as required in the constructor.

**Pull request closed without merging**